### PR TITLE
Options validation changes for admin and mongoclient

### PR DESCRIPTION
**Pull request by daprahamian**

### CODE REVIEW

**Comment by mbroadst** [comment]
@daprahamian ping, please remove bson-related options.

**Comment by daprahamian** [comment]
@kvwalker removed bson options!

File: lib/admin.js [hunk 1]
```diff
@@ -110,7 +119,12 @@ Admin.prototype.buildInfo = function(options, callback) {
  * instance of the db client
  *
  * @param {Object} [options] optional parameters for this operation
- * @param {ClientSession} [options.session] optional session to use for this operation
+ * @param {boolean} [options.raw=false] Return a raw nodejs Buffer instead of deserializing BSON
```
Comment (kvwalker) [inline lib/admin.js]: Can we remove `raw`, `fullResult`, and `serializeFunctions`?

File: lib/admin.js [hunk 2]
```diff
@@ -88,7 +92,12 @@ Admin.prototype.command = function(command, options, callback) {
...
```
Comment (kvwalker) [inline lib/admin.js]: Can we remove `raw`, `fullResult`, and `serializeFunctions`?

File: lib/admin.js [hunk 3]
```diff
@@ -65,8 +65,12 @@ function Admin(db, topology, promiseLibrary) {
  * @method
  * @param {object} command The command hash
  * @param {object} [options] Optional settings.
- * @param {(ReadPreference|string)} [options.readPreference] The preferred read preference (ReadPreference.PRIMARY, ReadPreference.PRIMARY_PREFERRED, ReadPreference.SECONDARY, ReadPreference.SECONDARY_PREFERRED, ReadPreference.NEAREST).
- * @param {number} [options.maxTimeMS] Number of milliseconds to wait before aborting the query.
+ * @param {boolean} [options.raw=false] Return a raw nodejs Buffer instead of deserializing BSON
```
Comment (kvwalker) [inline lib/admin.js]: Can we remove `raw`, `fullResult`, and `serializeFunctions`?

Comment (kvwalker) [reply lib/admin.js]: @daprahamian it looks like there are still a few bson options here.

**Pull request merged**